### ISSUE TITLE
Sometimes autocompletion doesn't work

### ISSUE THREAD

**Issue by KOLANICH** [ipython/ipython#11798]
```python
In [1]: import numpy
In [2]: import IPython
In [3]: ipy = IPython.get_ipython()
In [4]: ipy.Completer.all_completions("numpy.")
Out[4]: []
```
Python 3.7.3 (default, Apr  3 2019, 05:39:12) 
IPython 7.6.0.dev 6883de9d84006578fe044ae4d99e7a6a8ca7f10c
Latest parso and jedi from git are used.
`ipy.Completer.all_completions("IPython.")` works.

**Comment by dickmao** [comment]
I am experiencing trouble with the latest jedi==0.14.0

**Comment by davidnero** [comment]
I have the same issue. Ipython 7.6.1 with jedi 0.14.1 won't autocomplete at the module level.

Ipython 6.5.0 works and Ipython 7.6.1 with jedi downgraded to 0.13.3 works (but seems more sluggish than 6.5.0).

**Comment by bhodorog** [comment]
I can confirm this one.  My reproduce scenario involves `sys` module from standard lib
```
$ ipython
Python 3.7.2 (default, Feb 18 2019, 15:16:17)
Type 'copyright', 'credits' or 'license' for more information
IPython 7.6.1 -- An enhanced Interactive Python. Type '?' for help.

In [1]: !pip list
Package          Version
---------------- -------
appnope          0.1.0
backcall         0.1.0
decorator        4.4.0
ipython          7.6.1
ipython-genutils 0.2.0
jedi             0.14.1
parso            0.5.1
pexpect          4.7.0
pickleshare      0.7.5
pip              19.1.1
prompt-toolkit   2.0.9
ptyprocess       0.6.0
Pygments         2.4.2
setuptools       41.0.1
six              1.12.0
traitlets        4.3.2
wcwidth          0.1.7
wheel            0.33.4

In [2]: import sys

In [3]: sys.<tab-completion-here-doesn't work>
```
`pip install ipython jedi >0.13` will exhibit the completion problem
`pip install ipython jedi <0.14` works just fine

I had thought it's a jedi only problem, but using jedi + python repl as [described by jedi docs](https://jedi.readthedocs.io/en/latest/docs/usage.html#module-jedi.api.replstartup) completion for sys works just fine.

P.S. I've tested on OSX with a cpython 3.7.2 build with pyenv (default options).

**Comment by maxwellmckinnon** [comment]
Also experiencing this on macOS:
ipython 7.6.1
jedi 0.14.0
python 3.7.3

```
In [1]: import numpy

In [2]: import IPython

In [3]: ipy = IPython.get_ipython()

In [4]: ipy.Completer.all_completions("numpy.")
Out[4]: []
```

**Comment by minrk** [comment]
Closing as a jedi issue in 0.14.0-0.14.1. Fix should be in 0.14.2.

To workaround while waiting for release, pin to previous jedi:

    pip install `jedi==0.13.*`

or use old-style completions:

    get_ipython().Completer.use_jedi = False

I've confirmed that https://github.com/davidhalter/jedi/pull/1370 fixes the issue.

**Comment by maxwellmckinnon** [comment]
This is broken again in
```
jedi==0.15.1
ipython==7.8.0
Python 3.7.4
Was this not fixed with an added unit test against this? :P
```
In [1]: import numpy as np
i
In [2]: import IPython

In [3]: ipy = IPython.get_ipython()

In [4]: ipy.Completer.all_completions("numpy.")
Out[4]: []
